ocaml/lambda/matching.ml

```diff
@@ -1185,11 +1205,24 @@ let what_is_first_case = what_is_cases ~skip_any:false
 
 let what_is_cases = what_is_cases ~skip_any:true
 
+type pm_free_variables =
```

Comment:
Can we just call these constructors `Known` / `Unknown`? If we don't get any principality warnings (and I suspect we won't), I'd prefer that.